File: 2periodo/Estudo/Desafios/Beecrowd/Criptografia/main.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

bool isLetra(char c){
    return (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z');
}

void cript(char input[]){
    char* resp = "";
    for(int i=0;i<strlen(input);i++){
        if(isLetra(input[i])){
            resp += (char)(input[i]+3);
        }else{
            resp += input[i];
        }
    }
}

int main(){
    int casos = 0;
    scanf(" %i", &casos);
    char* input;
    for(int i=0;i<casos;i++){
        scanf(" %[^\n]", input);
        cript(input);
        printf("%s\n", input);
    }
    return 0;
}

File: 2periodo/Estudo/Desafios/Beecrowd/Criptografia/Main.java

import java.util.Scanner;

class Main{
    public static Scanner sc;

    public static boolean isLetra(char x){
        return (x >= 'a' && x <= 'z') || (x >= 'A' && x <= 'Z');
    }

    public static void cript3(String x){
        String resp = "";
        for(int i=0;i<(x.length()/2);i++){
            resp += (char)(x.charAt(i));
        }
        for(int i=(x.length()/2);i<x.length();i++){
            resp += (char)(x.charAt(i) - 1);
        }
        System.out.println(resp);
    }

    public static void cript2(String x){
        String resp = "";
        for(int i = x.length()-1;i>=0;i--){
            resp += x.charAt(i);
        }
        cript3(resp);
    }

    public static void cript(String x){
        String resp = "";
        for(int i=0;i<x.length();i++){
            if(isLetra(x.charAt(i))){
                resp += (char) (x.charAt(i)+3);
            } else{
                resp += x.charAt(i);
            }
        }
        cript2(resp);
    }

    public static void main(String[] args){ 
        sc = new Scanner(System.in);
        int casos = sc.nextInt();
        String x="";
        int i=0;
        sc.nextLine();
        while(i < casos){
            x = sc.nextLine();
            cript(x);
            i++;
        }
    }
}